controllers/useraccount/useraccount_controller_test.go

```diff
@@ -825,58 +952,69 @@ func TestReconcile(t *testing.T) {
 		assert.Equal(t, reconcile.Result{}, res)
 		require.EqualError(t, err, fmt.Sprintf("unable to add finalizer for user account %s", userAcc.Name))
 	})
+
 	// Remove finalizer fails
 	t.Run("remove finalizer fails", func(t *testing.T) {
 		// given
-		userAcc := newUserAccount(username, userID, false)
+		userAcc := newUserAccount(username, userID)
 		r, req, fakeClient, _ := prepareReconcile(t, username, userAcc, preexistingUser, preexistingIdentity)
 
-		//when
+		// when
 		res, err := r.Reconcile(context.TODO(), req)
 		require.NoError(t, err)
 		assert.Equal(t, reconcile.Result{}, res)
 
-		//then
-		userAcc = &toolchainv1alpha1.UserAccount{}
-		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: username, Namespace: test.MemberOperatorNs}, userAcc)
-		require.NoError(t, err)
-
+		// then
 		// Check that the finalizer is present
-		require.True(t, util.HasFinalizer(userAcc, toolchainv1alpha1.FinalizerName))
+		userAcc = useraccount.AssertThatUserAccount(t, username, r.Client).
+			HasFinalizer(toolchainv1alpha1.FinalizerName).
+			Get()
 
 		// Set the deletionTimestamp
-		userAcc.DeletionTimestamp = &metav1.Time{time.Now()} //nolint: govet
+		now := metav1.NewTime(time.Now())
+		userAcc.DeletionTimestamp = &now
 		err = r.Client.Update(context.TODO(), userAcc)
 		require.NoError(t, err)
 
-		// Mock finalizer removal failure
-		fakeClient.MockUpdate = func(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
-			finalizers := []string{toolchainv1alpha1.FinalizerName}
-			userAcc := obj.(*toolchainv1alpha1.UserAccount)
-			userAcc.Finalizers = finalizers
-			return fmt.Errorf("unable to remove finalizer for user account %s", userAcc.Name)
-		}
-
+		// trigger the deletion of the `User` resource
+		t.Log("Reconcile #1 with Deletion timestamp: deleting the User resource")
```

Comment:
I would rather put it into subtests. It's easier to maintain and better to understand.

Reply:
good point! Done in 4baa36f